### PR TITLE
Fixes #12202: Missing information/bad UX in API authorization UI 

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/12202

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-rest/src/main/scala/com/normation/rudder/rest/internal/RestApiAccounts.scala [hunk 1]
```diff
@@ -198,6 +197,7 @@ case class RestApiAccount(
       case x => x
     }
 
+    println(expiration)
```
Comment (fanf) [inline rudder-rest/src/main/scala/com/normation/rudder/rest/internal/RestApiAccounts.scala]: please remove it

File: rudder-web/src/main/webapp/javascript/rudder/angular/apiAccount.js [hunk 2]
```diff
@@ -91,23 +94,45 @@ accountManagement.controller('AccountCtrl', function ($scope, $http, DTOptionsBu
       $('#oldAccountPopup').bsModal('hide');
   }
 
-
-$scope.addAccount = function() {
-  var newAccount = { id : "", name : "", token: undefined, enabled : true, description : ""}
+  $scope.filterAccount = function (value,index,array) {
+	console.log($scope.authFilter)
```
Comment (fanf) [inline rudder-web/src/main/webapp/javascript/rudder/angular/apiAccount.js]: you should remove all the console.log

File: rudder-web/src/main/webapp/javascript/rudder/angular/apiAccount.js [hunk 3]
```diff
@@ -91,23 +94,45 @@ accountManagement.controller('AccountCtrl', function ($scope, $http, DTOptionsBu
...
+      return true;
+    } else {
+      return value.authorizationType === $scope.authFilter;
+    }
+  }
+  $scope.addAccount = function() {
+  var now = new Date();
+  now.setMonth(now.getMonth()+1);
+  var now = now.getFullYear()+"-"+(now.getMonth()+1)+"-"+now.getDate()+" "+now.getHours()+":"+now.getMinutes();
+  console.log(now);
```
Comment (fanf) [inline rudder-web/src/main/webapp/javascript/rudder/angular/apiAccount.js]: also here

File: rudder-web/src/main/webapp/javascript/rudder/angular/apiAccount.js [hunk 4]
```diff
@@ -137,6 +162,7 @@ $scope.popupDeletion = function(account, action, actionName) {
    }
  }
  $scope.saveAccount = function(account,isPopup) {
+console.log(account);
```
Comment (fanf) [inline rudder-web/src/main/webapp/javascript/rudder/angular/apiAccount.js]: this one

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by fanf** [comment]
OK to merge when the log are removed. It will still need woks because of the error on save which imply double save. 
Also, we don't get any message telling us that we can't enable an user account without the plugin, and we will need something in the UI to differenciate between the different kind of token

**Pull request merged**